spp_import_match/models/import_match.py

```diff
@@ -0,0 +1,127 @@
+
+
+class SPPImportMatch(models.Model):
+    _name = "spp.import.match"
+    _description = "Import Matching"
+    _order = "sequence, name"
+
+    name = fields.Char(compute="_compute_name", store=True, index=True)
+    sequence = fields.Integer(index=True)
+    model_id = fields.Many2one(
+        "ir.model",
+        "Model",
+        required=True,
+        ondelete="cascade",
+        domain=[("transient", "=", False)],
+        help="Model for Import Matching",
+    )
+    model_name = fields.Char(related="model_id.model")
+    model_description = fields.Char(related="model_id.name")
+    field_ids = fields.One2many(
+        "spp.import.match.fields",
+        "match_id",
+        string="Fields",
+        required=True,
+        help="Fields to Match in Importing",
+    )
+
+    @api.onchange("model_id")
+    def _onchange_model_id(self):
+        for rec in self:
+            rec.field_ids = None
+
+    @api.depends("model_id")
+    def _compute_name(self):
+        for rec in self:
+            name = "New"
+            if rec.model_id:
+                name = rec.model_description
+
+            rec.name = name
+
+    @api.model
+    def _match_find(self, model, converted_row, imported_row):
+        usable, field_to_match = self._usable_rules(model._name, converted_row)
+        usable = self.browse(usable)
+        for combination in usable:
+            combination_valid = True
+            domain = list()
+            for field in combination.field_ids:
+                if field.conditional:
+                    if imported_row[field.name] != field.imported_value:
+                        combination_valid = False
+                        break
+                if field.field_id.name in converted_row:
+                    row_value = converted_row[field.field_id.name]
+                    field_value = field.field_id.name
+                    add_to_domain = True
+                    if field.sub_field_id:
+                        tuple_val = row_value[0][2]
+                        add_to_domain = False
+                        if field.sub_field_id.name in tuple_val:
+                            row_value = tuple_val[field.sub_field_id.name]
+                            add_to_domain = True
+                            field_value = (
+                                field.field_id.name + "." + field.sub_field_id.name
+                            )
+                    if add_to_domain:
+                        domain.append((field_value, "=", row_value))
+            if not combination_valid:
+                continue
+            match = model.search(domain)
+            if len(match) == 1:
+                return match
+
+        return model
+
+    @api.model
+    @tools.ormcache("model_name", "frozenset(fields)")
+    def _usable_rules(self, model_name, fields):
+        result = self
+        available = self.search([("model_name", "=", model_name)])
+        field_to_match = []
+        for record in available:
```

Comment:
so if we have 2 sql constraints as I suggested, this function will be search with limit=1, save 1 for loop